smmips/smmips.py

```diff
@@ -113,23 +279,141 @@ def count_variants(bamfile, panel, outdir, max_depth, truncate, ignore_orphans,
+#            print('#############\n')
+#            print('AttributeError: {0}\n'.format(e))
+#            print('#############\n\n')
+#            print(parser.format_help())
+#    elif args.subparser_name == 'variant':
+#        try:
+#            count_variants(args.bamfile, args.panel, args.outdir, args.max_depth, args.truncate, args.ignore_orphans,
+#                   args.stepper, args.prefix, args.reference, args.cosmicfile)
+#        except AttributeError as e:
+#            print('#############\n')
+#            print('AttributeError: {0}\n'.format(e))
+#            print('#############\n\n')
+#            print(parser.format_help())
+#    elif args.subparser_name is None:
+#        print(parser.format_help())
+
+
+
+
+if __name__ == '__main__':
+       
     # create main parser    
     parser = argparse.ArgumentParser(prog='smmip.py', description="A tool to analyse smMIP libraries")
     subparsers = parser.add_subparsers(help='sub-command help', dest='subparser_name')
        		
+    
+    # align reads
+    al_parser = subparsers.add_parser('align', help='Align reads to reference genome')
+    al_parser.add_argument('-f1', '--Fastq1', dest='fastq1', help = 'Path to Fastq1', required=True)
+    al_parser.add_argument('-f2', '--Fastq2', dest='fastq2', help = 'Path to Fastq2', required=True)
+    al_parser.add_argument('-o', '--Outdir', dest='outdir', help = 'Path to outputd directory. Current directory if not provided')
+    al_parser.add_argument('-r', '--Reference', dest='reference', help = 'Path to the reference genome', required=True)
+    al_parser.add_argument('-bwa', '--Bwa', dest='bwa', help = 'Path to the bwa script', required=True)
+    al_parser.add_argument('--remove', dest='remove', action='store_true', help = 'Remove intermediate files. Default is False, becomes True if used')
+    al_parser.add_argument('-pf', '--Prefix', dest='prefix', help = 'Prefix used to name the output files', required=True)
+    
+    
     # assign smMips to reads
     a_parser = subparsers.add_parser('assign', help='Extract UMIs from reads and assign reads to smmips')
-    a_parser.add_argument('-f1', '--Fastq1', dest='fastq1', help = 'Path to Fastq1', required=True)
-    a_parser.add_argument('-f2', '--Fastq2', dest='fastq2', help = 'Path to Fastq2', required=True)
     a_parser.add_argument('-pa', '--Panel', dest='panel', help = 'Path to panel file with smmip information', required=True)
     a_parser.add_argument('-o', '--Outdir', dest='outdir', help = 'Path to outputd directory. Current directory if not provided')
-    a_parser.add_argument('-r', '--Reference', dest='reference', help = 'Path to the reference genome', required=True)
-    a_parser.add_argument('-bwa', '--Bwa', dest='bwa', help = 'Path to the bwa script', required=True)
+    a_parser.add_argument('-c', '--Chromosome', dest='chromosome', default='all', help = 'Considers only the reads mapped to chromosome. Default is all chromosomes. Accepted values are "all", "chrA"')
```

Comment:
i think this addresses my earlier comments. 
the tool can be given a -c argument, and it will process only a single chromosome.  logic for splitting by chromosome will be outside of this tool
if no chromosome is indicated, the it will process all.

Reply:
that's correct. the python script's parameter -c is given either a chromosome or "all" to indicate assigning reads mapped to that particular chromosome or to all chromosomes. The default value is "all" so omitting -c is equivalent to look at all chromosomes sequentially. Splitting by chromosomes is done at the workflow level.